backend/src/models/userCreds.js

```diff
@@ -9,8 +10,49 @@ const userCredsSchema = mongoose.Schema(
     refreshToken: { type: String },
     userDataId: { type: ObjectId }
   },
-  { timestamps: true }
+  {
+    timestamps: true
+    // toJSON: {
+    //   getters: true,
+    //   virtuals: true
+    // }
+  }
 );
 
+userCredsSchema.statics.upsertGoogleUser = function(accessToken, refreshToken, profile, cb) {
+  const that = this;
+  return this.findOne(
+    {
+      googleId: profile.id
+    },
+    (err, user) => {
+      // no user was found, lets create a new one
+      if (!user) {
+        const { email, given_name, family_name } = profile._json;
+        const newUserData = new UserData({
+          role: 'new',
+          bio: { email, first_name: given_name, last_name: family_name }
+        });
+
+        newUserData.save((err1, savedUserData) => {
+          const newUser = new that({
+            googleId: profile.id,
+            accessToken,
+            refreshToken,
+            userDataId: savedUserData.id
+          });
+          newUser.save((error, savedUser) => {
+            if (error) console.log(error);
```

Comment:
No need for `console.log`, all errors are logged by the express error handler.

Reply:
Do I need to do anything specifically to make sure the express error handler gets this error? I'm not sure I understand how it works.

Reply:
I don't think so. When the callback is called with an `err`, passport should automatically pass the `err` to the [error handler](https://github.com/GTBitsOfGood/drawchange/blob/f5a999cfc1b3ec961907cbf8f507941571cebd3e/backend/src/routes/index.js#L19). I've only tested this with MongoDB errors, but passport errors _should_ behave the same...